### PR TITLE
Add warning to master branch

### LINKED ISSUE #231: Switching `master` to `main`
Would be good to switch `master` to `main` on this repo.

As some users pin to `@master` in workflows, it probably can't be removed, but it would be good to include a deprecation notice with `@master` (maybe just a message that gets reverted in `main`).

### PR DESCRIPTION
**Pull request by goanpeca**
Fixes #231

Added a warning for the master branch on this PR @jakirkham 

Screen Shot 2022-05-26 at 12 01 04 PM

---

@conda-incubator/setup-miniconda thoughts?

### CODE REVIEW

**Comment by jakirkham** [comment]
LGTM. Thanks Gonzalo! 🙏

Will leave the more detailed review bits to others who know more 🙂

**Comment by jakirkham** [comment]
Thoughts @conda-incubator/setup-miniconda ? 🙂

File: dist/setup/index.js [hunk 1]
```diff
@@ -30000,7 +30008,7 @@ module.exports.safeLoad    = safeLoad;
...
...c":null,"fetchSpec":"1.0.0-rc.3"},"_requiredBy":["/get-hrefs"],"_resolved":"https://registry.npmjs.org/cheerio/-/cheerio-1.0.0-rc.3.tgz","_spec":"1.0.0-rc.3","_where":"/Users/gpenacastellanos/Dropbox (Personal)/develop/conda-incubator/setup-miniconda","author":{"name":"Matt Mueller","email":"mattmuelle@gmail.com","url":"mat.io"},"bugs":{"url":"https://github.com/cheeriojs/cheerio/issues"},"dependencies":{"css-select":"~1.2.0","dom-serializer":"~0.1.1","entities":"~1.1.1","htmlparser2":"^3.9.1","lodash":"^4.15.0","parse5":"^3.0.1"},"description":"Tiny, fast, and elegant implementation of core jQuery designed specifically for the server","devDependencies":{"benchmark":"^2.1.0","coveralls":"^2.11.9","expect.js":"~0.3.1","istanbul":"^0.4.3","jquery":"^3.0.0","jsdom":"^9.2.1","jshint":"^2.9.2","mocha":"^3.1.2","xyz":"~1.1.0"},"engines":{"node":">= 0.6"},"files":["index.js","lib"],"homepage":"https://github.com/cheeriojs/cheerio#readme","keywords":["htmlparser","jquery","selector","scraper","parser","html"],"license":"MIT","main":"./index.js","name":"cheerio","repository":{"type":"git","url":"git://github.com/cheeriojs/cheerio.git"},"scripts":{"test":"make test"},"version":"1.0.0-rc.3"};
```
Comment (mattwthompson) [inline dist/setup/index.js:30011]: I don't know TypeScript but this looks like a local path, which perhaps was not intended?

File: dist/setup/index.js [hunk 2]
```diff
@@ -20607,6 +20608,13 @@ function setupMiniconda(inputs) {
 function run() {
     return __awaiter(this, void 0, void 0, function* () {
         try {
+            core.warning(`
+  The 'master' branch is deprecated and will be removed from the repository in the future.
+  
+  Please ensure your action is using the latest version, or point it to the 'main' branch.
+
+    "conda-incubator/setup-minicando@main"
```
Comment (mattwthompson) [inline dist/setup/index.js:20616]:
```suggestion
    "conda-incubator/setup-miniconda@main"
```

File: src/setup.ts [hunk 3]
```diff
@@ -110,6 +110,13 @@ async function setupMiniconda(inputs: types.IActionInputs): Promise<void> {
  */
 async function run(): Promise<void> {
   try {
+    core.warning(`
+  The 'master' branch is deprecated and will be removed from the repository in the future.
+  
+  Please ensure your action is using the latest version, or point it to the 'main' branch.
+
+    "conda-incubator/setup-minicando@main"
```
Comment (mattwthompson) [inline src/setup.ts:118]:
```suggestion
    "conda-incubator/setup-miniconda@main"
```

**Review by mattwthompson — COMMENTED** [review]
I found a couple of typos, otherwise seems fine to somebody not familiar with the language. I don't think any of my projects point to the `master` branch, but that wouldn't matter for testing purposes as this is not in production. In principle, I think an action could point to an arbitrary feature branch, but I don't think that level of testing is necessary here.

AFAICT the changes in the README are only whitepsace/wrapping?

File: dist/setup/index.js (hunk 1 above)
Comment (goanpeca) [reply dist/setup/index.js:30011]: Interesting! haha, will check. But probably should not affect things.

**Pull request merged**